SSA/Projects/FullyHomomorphicEncryption/Basic.lean

```diff
@@ -485,11 +495,57 @@ theorem R.fromTensor_eq_fromTensor'_fromPoly {q n} : R.fromTensor (q := q) (n :=
       simp[monomial]
 
 
-/-- an equivalent implementation of `fromTensor` that uses `Finsupp` to enable reasoning about values using mathlib's notions of
+/-- an equivalent implementation of `fromTensor` that uses `Finsupp`
+  to enable reasoning about values using mathlib's notions of
   support, coefficients, etc. -/
 noncomputable def R.fromTensorFinsupp (coeffs : List ℤ) : (ZMod q)[X] :=
   Polynomial.ofFinsupp (List.toFinsupp (coeffs.map Int.cast))
 
+theorem Polynomial.degree_toFinsupp [Semiring M] [DecidableEq M]
+  (xs : List M) :
+  degree { toFinsupp := List.toFinsupp (l := xs) } ≤ List.length xs := by
+    cases xs
+    case nil =>
+      simp[degree]
```

Comment:
nit: whitespace (and all the others below)

Reply:
@goens what exactly is the nit?

Reply:
oh I see between the parens, OK